### PR TITLE
Remove lodash from calypso-analytics

### PR DESCRIPTION
**Pull request by sgomes**
`calypso-analytics` currently depends on `lodash` for some relatively trivial usage.

This PR removes that dependency, replacing the `lodash` usage with vanilla JS.

CC @Copons and @Automattic/t-rex, who are aware of me working on this change. 

## Proposed Changes

* Remove `lodash` from `calypso-analytics`

## Testing Instructions

I'm not entirely sure what the best way of testing this is. Some simple smoke-testing seems to indicate that events are still being sent. Is that enough?

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-analytics/src/tracks.ts [hunk 1]
```diff
@@ -177,7 +178,7 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 	if ( process.env.NODE_ENV !== 'production' && typeof console !== 'undefined' ) {
 		if (
 			! /^calypso(?:_[a-z0-9]+){2,}$/.test( eventName ) &&
-			! includes( EVENT_NAME_EXCEPTIONS, eventName )
+			! EVENT_NAME_EXCEPTIONS.includes( eventName )
```
Comment (sgomes) [inline packages/calypso-analytics/src/tracks.ts:181]: This is a non-empty array, defined at the top of the file.

File: packages/calypso-analytics/src/tracks.ts [hunk 2]
```diff
@@ -221,7 +222,7 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 
 	debug( 'Record event "%s" called with props %o', eventName, eventProperties );
 
-	if ( ! eventName.startsWith( 'calypso_' ) && ! includes( EVENT_NAME_EXCEPTIONS, eventName ) ) {
+	if ( ! eventName.startsWith( 'calypso_' ) && ! EVENT_NAME_EXCEPTIONS.includes( eventName ) ) {
```
Comment (sgomes) [inline packages/calypso-analytics/src/tracks.ts:225]: This is a non-empty array, defined at the top of the file.

File: packages/calypso-analytics/src/tracks.ts [hunk 3]
```diff
@@ -233,7 +234,9 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 
 	// Remove properties that have an undefined value
 	// This allows a caller to easily remove properties from the recorded set by setting them to undefined
-	eventProperties = omitBy( eventProperties, ( prop ) => typeof prop === 'undefined' );
+	eventProperties = Object.fromEntries(
+		Object.entries( eventProperties ).filter( ( [ , val ] ) => typeof val !== 'undefined' )
```
Comment (sgomes) [inline packages/calypso-analytics/src/tracks.ts:238]: The code ensures `eventProperties` is not empty at the beginning of the function.

**Review by tyxla — APPROVED** [review]
The code looks great and events are still being sent correctly. 👍 

Thanks @sgomes 🙌 🚀

**Comment by sgomes** [comment]
Thank you for the review, @tyxla! 👍

**Pull request merged**